### PR TITLE
added focal_plane nan checker in coords.pmat.for_tod

### PR DESCRIPTION
**Pull request by tterasaki**
https://github.com/simonsobs/sotodlib/issues/763

### CODE REVIEW

File: sotodlib/coords/pmat.py [hunk 1]
```diff
@@ -189,6 +189,9 @@ def for_tod(cls, tod, sight=None, geom=None, comps='T',
         # Set up the detectors in the focalplane
         fp = _valid_arg(focal_plane, 'focal_plane', src=tod)
         xi, eta, gamma  = fp.xi, fp.eta, fp.get('gamma')
+        if not np.all(~np.isnan([xi, eta, gamma])):
+            raise ValueError('np.nan is included in focal_plane')
+
```
Comment (mhasself) [inline sotodlib/coords/pmat.py:194]: gamma is allowed to be None here, which I think will cause the array promotion of [xi, eta, gamma] to fail.

Simpler logic uses `if np.any(np.isnan(...))`.

Comment (mhasself) [inline sotodlib/coords/pmat.py:194]: gamma is allowed to be None here, which I think will cause the array promotion of [xi, eta, gamma] to fail.

Simpler logic uses `if np.any(np.isnan(...))`.

**Review by mhasself — CHANGES_REQUESTED** [review]

**Comment by tterasaki** [comment]
Thanks Matthew and sorry for the late reply.
I think it is ready to merge.

**Comment by mhasself** [comment]
> Thanks Matthew and sorry for the late reply. I think it is ready to merge.

No prob -- but see my comment about "gamma".  It can be None.  This code will choke on that.

**Comment by kmharrington** [comment]
When will this converge for merging? Enough other people are hitting this issue and solving it (ex: https://github.com/simonsobs/sotodlib/pull/816 ) that we need to have the nice version in master

**Comment by mhasself** [comment]
I am sorry this is so much fun, but I think you need to also add this:
```
        if gamma is not None and np.any(np.isnan(gamma)):
            raise ValueError('np.nan is included in gamma')
```

**Review by mhasself — APPROVED** [review]
To infinity and beyond!

**Pull request merged**